ambry-store/src/test/java/com.github.ambry.store/StoreDescriptorTest.java

```diff
@@ -0,0 +1,89 @@
+   * @throws IOException
+   */
+  @Test
+  public void testStoreDescriptor() throws IOException {
+    String tempDirPath = "/tmp/";
+    String storeId = "demoStoreId";
+    StoreDescriptor storeDescriptor = new StoreDescriptor(tempDirPath, storeId);
+    // store descriptor file should have been created.
+    StoreDescriptor newStoreDescriptor = new StoreDescriptor(tempDirPath, storeId);
+    Assert.assertEquals("StoreId mismatch ", storeDescriptor.getStoreId(), newStoreDescriptor.getStoreId());
+    Assert.assertEquals("IncarnationId mismatch ", storeDescriptor.getIncarnationId(),
+        newStoreDescriptor.getIncarnationId());
+
+    // read the file to fetch storeId and incarnationId
+    newStoreDescriptor = StoreDescriptor.fromBytes(
+        new DataInputStream(new FileInputStream(new File(tempDirPath, StoreDescriptor.STORE_DESCRIPTOR))));
+    Assert.assertEquals("StoreId mismatch ", storeDescriptor.getStoreId(), newStoreDescriptor.getStoreId());
+    Assert.assertEquals("IncarnationId mismatch ", storeDescriptor.getIncarnationId(),
+        newStoreDescriptor.getIncarnationId());
+
+    // Create StoreDescriptor file with explicit store Id and incarnationId
+    storeId = "demoStoreId2";
+    UUID incarnationId = UUID.randomUUID();
+    int size =
+        StoreDescriptor.VERSION_SIZE + StoreDescriptor.STORE_ID_LENGTH_SIZE + storeId.toString().getBytes().length +
+            StoreDescriptor.INCARNATION_ID_LENGTH_SIZE + incarnationId.toString().getBytes().length;
+    byte[] toBytes = new byte[size];
+    ByteBuffer byteBuffer = ByteBuffer.wrap(toBytes);
+    byteBuffer.putShort(StoreDescriptor.VERSION_0);
+    byteBuffer.putInt(storeId.toString().getBytes().length);
+    byteBuffer.put(storeId.toString().getBytes());
+    byteBuffer.putInt(incarnationId.toString().getBytes().length);
+    byteBuffer.put(incarnationId.toString().getBytes());
+    byteBuffer.flip();
+    storeDescriptor = StoreDescriptor.fromBytes(new DataInputStream(new ByteBufferInputStream(byteBuffer)));
+    Assert.assertEquals("StoreId mismatch ", storeId, storeDescriptor.getStoreId());
+    Assert.assertEquals("IncarnationId mismatch ", incarnationId, storeDescriptor.getIncarnationId());
+
+    // version mismatch
+    size = StoreDescriptor.VERSION_SIZE + StoreDescriptor.STORE_ID_LENGTH_SIZE + storeId.toString().getBytes().length +
+        StoreDescriptor.INCARNATION_ID_LENGTH_SIZE + incarnationId.toString().getBytes().length;
+    toBytes = new byte[size];
+    byteBuffer = ByteBuffer.wrap(toBytes);
+    byteBuffer.putShort((short) 2);
+    byteBuffer.putInt(storeId.toString().getBytes().length);
+    byteBuffer.put(storeId.toString().getBytes());
+    byteBuffer.putInt(incarnationId.toString().getBytes().length);
+    byteBuffer.put(incarnationId.toString().getBytes());
+    byteBuffer.flip();
+
+    try {
+      StoreDescriptor.fromBytes(new DataInputStream(new ByteBufferInputStream(byteBuffer)));
+    } catch (IllegalArgumentException e) {
+    }
+  }
```

Comment:
Once you deal with store id mismatch in the main code, you should add appropriate tests.